araproc/analysis/correlation_snr.py

```diff
@@ -0,0 +1,254 @@
+    pairs_V = list(itertools.combinations(rf_chs[:8], 2))
+    pairs_H = list(itertools.combinations(rf_chs[8:], 2))
+
+    return pairs, pairs_V, pairs_H
+
+
+
+def pairing(channels):
+    
+    """
+    Makes channel pairs.
+    
+    Parameters
+    ----------
+    channels: list
+       List of given channels.
+
+    Returns
+    -------
+    pairs: list
+       List of all possible pair combinations of given channels.
+    """
+ 
+    pairs = list(itertools.combinations(channels, 2))
+    
+    return pairs
+
+
+
+def get_corr(y1, y2):
+    
+    """
+    Calculates normalized correlation between two waveforms, y1 and y2.
+    
+    Parameters
+    ----------
+    y1: list
+       List of values of waveform 1.
+    y2: list
+       List of values of waveform 2.
+    
+    Returns
+    -------
+    corr: array
+       Array of values of cross correlation function.  
+    """
+
+    _, y1 = wfu.tgraph_to_arrays(y1)
+    _, y2 = wfu.tgraph_to_arrays(y2)
+    
+    n = len(y1)
+    corr = co(y2, y1, mode = 'same')/np.sqrt(co(y1, y1, mode = 'same')[int(n/2)]*co(y2, y2, mode = 'same')[int(n/2)])    
+
+    return corr
+
+
+
+def get_corr_snr(first_ch, second_ch):
+
+    """
+    Calculates channel-pair correlation SNR
+    
+    Parameters
+    ----------
+    first_ch: array 
+       Array of a given channel waveform.
+    second_ch: array 
+       Array of another given channel waveform.
+     
+    Returns
+    -------
+    corr_snr: float
+       Channel pair correlation SNR.
+    """
+    
+    corr = get_corr(first_ch, second_ch)
+    corr_snr = snr.get_snr(corr)
+
+    return corr_snr
+
+
+def get_avg_corr_snr(wave_bundle, excluded_channels = []):
+
+    """
+    Makes useful dictionaries for channel-pair correlation SNR and their average.
+
+    Parameters
+    ----------
+    wave_bundle: dict of TGraphs
+        Dictionary of waveform TGraphs to be averaged.
+    excluded_channels: list
+        List of dictionary keys to exclude from average.
+
+    Returns
+    -------
+    avg_corr_snr : float
+        Average channel-pair correlation SNR.
+    avg_vpol_corr_snr : float
+        The VPol average channel-pair correlation SNR.
+    avg_hpol_corr_snr : float
+        The HPol average channel-pair correlation SNR.
+    """
+
+
+    chans = list(wave_bundle.keys())
+
+    AllPol_dict, VPol_dict, HPol_dict = {}, {}, {}
+    corr_snr, VPol_corr_snr, HPol_corr_snr = {}, {}, {}
+
+    for chan in chans:
+
+      if(chan in excluded_channels):
+        continue
+
+      AllPol_dict.update({chan: wave_bundle[chan]})
+
+      if chan in chans[:8]:
+        VPol_dict.update({chan: wave_bundle[chan]})
+      if chan in chans[8:]:
+        HPol_dict.update({chan: wave_bundle[chan]})
+    
+    avg_corr_snr, avg_vpol_corr_snr, avg_hpol_corr_snr = [], [], []
+
+    pairs = pairing(list(AllPol_dict.keys()))
+    pairs_V = pairing(list(VPol_dict.keys()))
+    pairs_H = pairing(list(HPol_dict.keys()))
+
+    for pair in pairs:
```

Comment:
May not matter once we switch to the pairs in standard_reco.py, but we should be careful pairs aren't double counted.